plugins/woocommerce/src/Internal/Admin/Orders/Edit.php

```diff
@@ -155,24 +156,28 @@ public function setup( \WC_Order $order ) {
 		$this->add_order_specific_meta_box();
 		$this->add_order_taxonomies_meta_box();
 
-		/**
-		 * From wp-admin/includes/meta-boxes.php.
-		 *
-		 * Fires after all built-in meta boxes have been added. Custom metaboxes may be enqueued here.
-		 *
-		 * @since 3.8.0.
-		 */
-		do_action( 'add_meta_boxes', $this->screen_id, $this->order );
+		$order_post = get_post( $this->order->get_id() );
 
-		/**
-		 * Provides an opportunity to inject custom meta boxes into the order editor screen. This
-		 * hook is an analog of `add_meta_boxes_<POST_TYPE>` as provided by WordPress core.
-		 *
-		 * @since 7.4.0
-		 *
-		 * @param WC_Order $order The order being edited.
-		 */
-		do_action( 'add_meta_boxes_' . $this->screen_id, $this->order );
+		if ( $order_post instanceof WP_Post ) {
+			/**
+			 * From wp-admin/includes/meta-boxes.php.
+			 *
+			 * Fires after all built-in meta boxes have been added. Custom metaboxes may be enqueued here.
+			 *
+			 * @since 3.8.0.
+			 */
+			do_action( 'add_meta_boxes', $this->screen_id, $order_post );
+
+			/**
+			 * Provides an opportunity to inject custom meta boxes into the order editor screen. This
+			 * hook is an analog of `add_meta_boxes_<POST_TYPE>` as provided by WordPress core.
+			 *
+			 * @since 7.4.0
+			 *
+			 * @param WP_Post $order_post The order being edited.
+			 */
+			do_action( 'add_meta_boxes_' . $this->screen_id, $order_post );
+		}
```

Comment:
Thanks for your patience, @amiut. This section all seems reasonable, but we think it makes sense to also add a new action *outside* of this block, that passes the order object. For example, `woocommerce_order_editor_add_meta_boxes`.

Reply:
Hello @barryhughes, thank you very much for your review.
Yes, that makes total sense to me.

I wanted to add below action after Line 180, but after checking other occurrences I realised they all have a `@since` in their dockblock, should I add it here is it ok to leave it empty? and what should version should it be if it's needed?

```PHP
/**
 * Provides an opportunity to inject custom meta boxes into the order editor screen.
 *
 * @param WC_Order $order The order being edited.
 */
do_action( 'woocommerce_order_editor_add_meta_boxes', $this->order );
```

Reply:
I believe `@since` is needed to pass linting (which is a required CI check). We can set it to `9.5.0`.

Reply:
Thank you, I added the hook.